### PR TITLE
fix: windows host process execution

### PR DESCRIPTION
**Pull request by wayfarer-rus**
When running under windows environment few characters messed up runtime

### CODE REVIEW

File: src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt [hunk 1]
```diff
@@ -128,8 +128,9 @@ class KotlinCompiler(
     outputDirectory: OutputDirectory,
     args: List<String>
   ): List<String> {
+    val pathSeparator = System.getProperty("path.separator") ?: ":"
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt:131]: Looks like a global constant

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt (hunk 1 above)
Comment (wayfarer-rus) [reply src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt:131]: where should I put it? 

your suggestion

Comment (AlexanderPrendota) [reply src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt:131]: private class field?

Comment (wayfarer-rus) [reply src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt]: done!

File: src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt [hunk 2]
```diff
@@ -148,4 +148,8 @@ class KotlinCompiler(
     }
   }
 
+  companion object {
+    private val PATH_SEPARATOR = System.getProperty("path.separator") ?: ":"
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/KotlinCompiler.kt:152]: please top of the class

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

**Comment by AlexanderPrendota** [comment]
@wayfarer-rus also maybe some test? =)

**Pull request merged**